tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -2462,6 +2695,758 @@ TEST_F(ExecutionTest, LifetimeIntrinsicTest) {
+        }
+    }
+    else if (payload.hitTest == AH_INDEX)
+    {
+        // Tested below since CH interferes with AH testing.
+    }
+    else
+    {
+        // Failure. No hit/miss shader found.
+        testBuffer[FAILED_INDEX] = 1;
+    }
+
+    // Reset and Trace without CH so that AH can be properly tested.
+    payload.color_read_write = float4(0.12, 0.34, 0.56, 0.78);
+    payload.color_read_only = float4(0.1, 0.2, 0.3, 0.4);
+    payload.color_write_only = float4(0,0,0,0);
+    payload.color_write_AH = float4(0.05, 0.04, 0.15, 0.28);
+    payload.hitTest = -1;
+    TraceRay(topObject, RAY_FLAG_SKIP_CLOSEST_HIT_SHADER, 0xFF, 0, 1, 0, ray, payload);
+
+    if (payload.hitTest == AH_INDEX)
+    {
+        // AH is expected to write, but not read 'color_read_write', thus color_write_AH should not match.
+        if (!all(payload.color_write_only == (WRITE_COLOR + payload.color_read_only)) ||
+            all(payload.color_write_AH == (WRITE_COLOR + payload.color_read_write)))
+        {
+            // Failure.
+            testBuffer[AH_INDEX] = 0;
+        }
+    }
+}
+
+[shader("miss")]
+void miss_read_write(inout PerRayData payload)
+{
+    payload.color_read_write = WRITE_COLOR;
+    payload.color_write_only = payload.color_read_write + payload.color_read_only;
+    // If 'color_read_only' could be written it would cause a failure in the RG check.
+    payload.color_read_only = WRITE_COLOR;
+    payload.hitTest = MS_INDEX;
+}
+
+[shader("anyhit")]
+void ah_write(inout PerRayData payload, in Attrs attrs)
+{
+    payload.color_write_only = WRITE_COLOR + payload.color_read_only;
+    // If 'color_read_write' could be read it would cause a failure in the RG check.
+    payload.color_write_AH = WRITE_COLOR + payload.color_read_write;
+    payload.hitTest = AH_INDEX;
+}
+
+[shader("closesthit")]
+void ch_read(inout PerRayData payload, in Attrs attrs)
+{
+    payload.color_write_only = payload.color_read_write + payload.color_read_only;
+    // If 'color_read_write' could be written it would cause a failure in the RG check.
+    payload.color_read_write = float4(0,0,0,0);
+    payload.hitTest = CH_INDEX;
+}
+
+)";
+
+  CComPtr<ID3D12Device> pDevice;
+  bool bSM_6_7_Supported = CreateDevice(&pDevice, D3D_SHADER_MODEL_6_7, false);
+  bool bSM_6_6_Supported = bSM_6_7_Supported;
+  if (!bSM_6_7_Supported) {
+    // Try 6.6 for downlevel
+    bSM_6_6_Supported = CreateDevice(&pDevice, D3D_SHADER_MODEL_6_6, false);
+  }
+  if (!bSM_6_6_Supported) {
+    WEX::Logging::Log::Comment(L"PAQTest requires shader model 6.6+ but no "
+                               L"supported device was found.");
+    WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+    return;
+  }
+  bool bDXRSupported =
+      bSM_6_6_Supported && DoesDeviceSupportRayTracing(pDevice);
+
+  if (GetTestParamUseWARP(UseWarpByDefault()) ||
+      IsDeviceBasicAdapter(pDevice)) {
+    WEX::Logging::Log::Comment(L"WARP does not support SM6.6+, required by "
```

Comment:
We've since deleted all of these checks and they shouldn't be added to new tests.